### PR TITLE
Expand system.accept_only_sharer pconfig value to include previous behavior with likes

### LINKED ISSUE #11245: Restore thread completion for Like activities behind a setting
From https://github.com/friendica/friendica/pull/11240#issuecomment-1040566477

> There is the boolean config variable `accept_only_sharer`. Every occurrence has to be replaced with a new configuration that has got with three options. Then the programming in this PR (in line 650) can be controlled by this new configuration as well.

### PR DESCRIPTION
**Pull request by MrPetovan**
Hopefully closes #11245

I found out that the existing but apparently unintended behavior with likes was only possible with AP since OStatus had a specific condition for it.

I kept the config key name which makes the resulting values unordered but with the constants it should hide the counter-intuitive values.

I also completely rewrote the setting help, let me know if it's unclear.

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -1403,7 +1408,7 @@ public static function storeForUserByUriId(int $uri_id, int $uid, array $fields
 		$is_reshare = ($item['gravity'] == GRAVITY_ACTIVITY) && ($item['verb'] == Activity::ANNOUNCE);
 
 		if ((($item['gravity'] == GRAVITY_PARENT) || $is_reshare) &&
-			DI::pConfig()->get($uid, 'system', 'accept_only_sharer') &&
+			DI::pConfig()->get($uid, 'system', 'accept_only_sharer') === self::COMPLETION_NONE &&
```
Comment (annando) [inline src/Model/Item.php:1411]: The pconfig value will most likely contain a boolean value, so comparing it via `===` wouldn't work, wouldn't it?

**Review by annando — COMMENTED** [review]
AFAIK the pconfig class does return config values always with the correct variable type. So shouldn't a comparison between a boolean value and an integer value alway return "false"?

File: src/Model/Item.php (hunk 1 above)
Comment (MrPetovan) [reply src/Model/Item.php:1411]: The pconfig value was stored as an integer even when it was a boolean flag, so there's no issue there, look at the assignation in `settings_post()`.

Comment (MrPetovan) [reply src/Model/Item.php:1411]: Ping @annando, I'm personally holding off on upgrading my node until this is merged not to lose the thread completion feature on likes.

Comment (annando) [reply src/Model/Item.php:1411]: Sorry, I was busy with the other task.

**Comment by tobiasd** [comment]
How can I disable the thread completion again? Before this PR was pulled onto my system everything was fine. Now the setting in the general social media section seems to have lost its meaning, even after re-setting it to "Only conversations my followed contacts started."

**Comment by MrPetovan** [comment]
You did the right thing picking this option, I thought I translated the option correctly but I couldn't really test it since it partly relies on incoming content from people I'm not following. I'll check again.

**Comment by MrPetovan** [comment]
It turns out @annando was right, if only partially. We can't use the strict comparison operator `===` with the integer constants I created because the value returned by `PConfig->get` is a string, and not a boolean like he suggested. I'll have a fix in a couple minutes.

**Pull request merged**